app/lib/widgets/cards/productCard/shared/badge.dart

```diff
@@ -0,0 +1,83 @@
+import 'package:flex_ui/tokens/sizes.dart';
+import 'package:flex_ui/utils/extensions.dart';
+import 'package:flutter/material.dart';
+import 'package:widgetbook/widgetbook.dart';
+import 'package:widgetbook_annotation/widgetbook_annotation.dart' as widgetbook;
+
+///A simple Badge Widget providing override properties for
+///
+/// - textColor
+/// - iconBackgroudnColor
+/// - padding
+/// - border radius
+///
+class FlexBadge extends StatelessWidget {
+  const FlexBadge({
+    super.key,
+    required this.label,
+    this.iconBackgroundColor,
+    this.textColor,
+    this.padding,
+    this.borderRadius,
+  });
+
+  final String label;
+  final Color? iconBackgroundColor;
+  final Color? textColor;
+  final EdgeInsets? padding;
+  final BorderRadius? borderRadius;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      decoration: BoxDecoration(
+        borderRadius:
+            borderRadius ?? BorderRadius.circular(FlexSizes.circleRadius),
+        color: iconBackgroundColor ?? context.colors.info,
+      ),
+      padding: padding ??
+          const EdgeInsets.symmetric(
+            horizontal: FlexSizes.md,
+            vertical: FlexSizes.xxs,
+          ),
+      child: Text(
+        label,
+        style: Theme.of(context)
+            .textTheme
+            .labelSmall
+            ?.copyWith(color: context.colors.onInfo)
+            .copyWith(color: textColor),
+      ),
+    );
+  }
+}
+
+@widgetbook.UseCase(
+  name: 'Default',
+  type: FlexBadge,
+  path: '[Components]',
+)
+Widget defaultBadge(BuildContext context) {
```

Comment:
how is defaultBadge used? Is it part of testing?

Reply:
Yea all the bottom stuff here is for Widgetbook, which just gets used for the dev demo library. 
 You can see them tagged with ``@widgetbook.UseCase`` just above (all ties back to the ``main.directories.g.dart`` code gen)